subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/RichVersionConstraintsIntegrationTest.groovy

```diff
@@ -462,4 +463,96 @@ class RichVersionConstraintsIntegrationTest extends AbstractModuleDependencyReso
             }
         }
     }
+
+
+    void "can reject dependency versions of an external component"() {
+        given:
+        repository {
+            'org:foo:1.0'()
+        }
+
+        buildFile << """
+            dependencies {
+                conf('org:foo:1.0') {
```

Comment:
:x: since the version here is `1.0` this test does not really test anything. Expect that the reject keyword can be parsed. I think we should use a version range that goes up to `1.1` here.

Reply:
That's precisely what this test is checking: that you can reject something. The fact that the semantics of rejecting are honored are in different tests.

Reply:
Ok.